### PR TITLE
feat: use postmark as email-provider

### LINKED ISSUE #451: Adress issue with SES not sending mails to everyone on high load
## Description

We're currently limited by AWS to 14 emails per second, where any emails beyond that point will not be sent. The exception is if we request to send e.g. 50 emails, then AWS will simply block any further requests until enough seconds have passed.

In order to alleviate this, we have requested an increase in our sending rate. Further efforts to mitigate this could include adding emails to some sort of queue, however, that relies on a task system, which is pending.

### LINKED ISSUE #529: Migrate from SES to Postmark

### PR DESCRIPTION
**Pull request by larwaa**
## Proposed Changes

- Migrate from AWS SES to [Postmark](https://postmarkapp.com) as e-mail provider
- Add wrapper classes for Postmark e-mails
- Migrate order confirmations and event e-mails to Postmark using Postmark templates
- Migrate Cabins to Postmark

## To do
- [x] Purchase a subscription
- [x] PR to refactor .env structure (#535)

## Types of Changes

- [x] New feature
- [ ] Bug fix
- [x] Enhancement/optimization
- [ ] Refactor
- [ ] Style
- [ ] Build/dependencies
- [ ] Other

## Issue(s) fixed or closed by this PR

- Resolves #529 
- Resolves #451 

## Checklist

- [ ] I have added tests to cover my changes (for features/bugs)
- [x] I am pleased with the readability of the code (if not, state where you need input)
- [x] I have tested the changes and verified that they work and do not break anything (to the best of my ability)

### CODE REVIEW

File: backend/.env.development [hunk 1]
```diff
@@ -11,9 +11,13 @@ AWS_SES_REGION_NAME=<INSERT AWS SES REGION NAME>
 DATAPORTEN_ID=fcaa9e30-a6d3-4809-8fea-cdd7b3de1c98
 DATAPORTEN_REDIRECT_URI=http://localhost:3000/authCallback
 DATAPORTEN_SECRET=<INSERT DATAPORTEN SECRET>
+DJANGO_BASE_URL=http://localhost:3000
```
Comment (Jorgro) [inline backend/.env.development:14]:
```suggestion
FRONTEND_BASE_URL=http://localhost:3000
```

File: backend/.env.example [hunk 2]
```diff
@@ -6,9 +6,13 @@ AWS_SES_REGION_NAME=<INSERT AWS SES REGION NAME>
 DATAPORTEN_ID=fcaa9e30-a6d3-4809-8fea-cdd7b3de1c98
 DATAPORTEN_REDIRECT_URI=http://localhost:3000/authCallback
 DATAPORTEN_SECRET=<INSERT DATAPORTEN SECRET>
+DJANGO_BASE_URL=http://localhost:3000
```
Comment (Jorgro) [inline backend/.env.example:9]:
```suggestion
FRONTEND_BASE_URL=http://localhost:3000
```

File: backend/.env.production [hunk 3]
```diff
@@ -11,6 +11,9 @@ CORS_ORIGIN_WHITELIST=https://www.indokntnu.no,https://indokntnu.no,https://call
 DATAPORTEN_ID=f17d2ea0-a7c9-4458-83bf-35cf5b555cae
 DATAPORTEN_REDIRECT_URI=https://indokntnu.no/authCallback
 DJANGO_ALLOWED_HOSTS=api.indokntnu.no,
+DJANGO_BASE_URL=https://indokntnu.no
```
Comment (Jorgro) [inline backend/.env.production:14]:
```suggestion
FRONTEND_BASE_URL=https://indokntnu.no
```

File: backend/.env.test [hunk 4]
```diff
@@ -18,10 +18,14 @@ DB_PASSWORD=postgres
 DB_PORT=5432
 DB_USER=postgres
 DJANGO_ALLOWED_HOSTS=0.0.0.0,localhost,backend
+DJANGO_BASE_URL=http://localhost:3000
```
Comment (Jorgro) [inline backend/.env.test:21]:
```suggestion
FRONTEND_BASE_URL=http://localhost:3000
```

File: backend/config/settings/base.py [hunk 5]
```diff
@@ -146,11 +146,18 @@
 CORS_ORIGIN_WHITELIST = env.list("CORS_ORIGIN_WHITELIST", default="http://localhost:3000")
 ALLOWED_HOSTS = env.list("DJANGO_ALLOWED_HOSTS", default=["localhost"])
 
+# URLS
+BASE_URL = env("DJANGO_BASE_URL", default="https://www.indokntnu.no")
```
Comment (Jorgro) [inline backend/config/settings/base.py:150]:
```suggestion
BASE_URL = env("FRONTEND_BASE_URL", default="https://www.indokntnu.no")
```

File: backend/config/settings/local.py [hunk 6]
```diff
@@ -12,6 +12,7 @@
 # URLs
 ROOT_URLCONF = "config.urls.local"
 ENVIRONMENT = env("DJANGO_ENVIRONMENT", default="development")
+BASE_URL = env("DJANGO_BASE_URL", default="http://localhost:3000")
```
Comment (Jorgro) [inline backend/config/settings/local.py:15]:
```suggestion
BASE_URL = env("FRONTEND_BASE_URL", default="http://localhost:3000")
```

File: backend/config/settings/base.py (hunk 5 above)
Comment (Jorgro) [inline backend/config/settings/base.py:150]: Consider changing `BASE_URL` to `FRONTEND_BASE_URL` too as it is a bit unclear what url this actually is.. What is the "base"?

File: backend/apps/ecommerce/mail.py [hunk 7]
```diff
@@ -27,30 +16,30 @@ def send_order_confirmation_mail(order: Order) -> None:
...
+    template_variables: TemplateVariables = {
+        user.email: {
+            "product_url": f"{settings.BASE_URL}/ecommerce/fallback?orderId={order.id}",
+            "product_name": order.product.name,
+            "name": user.first_name,
+            "receipt_id": str(order.id),
+            "date": order.timestamp.strftime("%d.%m.%Y, %X"),
+            "receipt_details": receipt_details,
+            "total": f"{order.total_price} NOK",
+            "support_mail": "kontakt@rubberdok.no",
```
Comment (Jorgro) [inline backend/apps/ecommerce/mail.py:34]: Should this be hardcoded in or be a variable someplace?

**Review by Jorgro — APPROVED** [review]
Would like you to improve naming of some environment variables, but approving since it is only minor so you don't have to wait for reapproval.

File: backend/apps/ecommerce/mail.py (hunk 7 above)
Comment (larwaa) [reply backend/apps/ecommerce/mail.py]: Added to env and moved to a `CONTACT_EMAIL` variable

File: backend/config/settings/base.py (hunk 5 above)
Comment (larwaa) [reply backend/config/settings/base.py]: Great suggestion, fixed in afe6e3f

**Pull request merged**